src/webgpu/api/validation/queue/copyToTexture/CopyExternalImageToTexture.spec.ts

```diff
@@ -323,9 +327,20 @@ g.test('source_image,crossOrigin')
       t.skip('DOM is not available to create an image element.');
     }
 
-    const crossOriginUrl = 'https://get.webgl.org/conformance-resources/opengl_logo.jpg';
-    const originCleanUrl = getResourcePath('Di-3d.png');
+    // Using 'localhost' and '127.0.0.1' trick to load cross origin resource. Set cross origin host name
+    // to 'localhost' if case is not running in 'localhost' domain. Otherwise, use '127.0.0.1'.
+    // host name to locahost unless the server running in
+    let crossOriginHostName = '';
+    if (location.hostname === 'localhost') {
+      crossOriginHostName = 'http://127.0.0.1';
+    } else {
+      crossOriginHostName = 'http://localhost';
+    }
```

Comment:
I think this logic should be in `resources.ts`, because the cross-origin hostnames won't be specific to this test.